File: backend_java/src/main/java/com/petconnect/model/AdoptionRequest.java
package com.petconnect.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Transient;

@Entity
@Table(name = "adoption_requests")
public class AdoptionRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long petId;
    private String requesterId;
    private String shelterOwnerId;
    private String status;

    @Transient
    private Pet pet;

    @Transient
    private User requester;

    public AdoptionRequest() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getPetId() {
        return petId;
    }

    public void setPetId(Long petId) {
        this.petId = petId;
    }

    public String getRequesterId() {
        return requesterId;
    }

    public void setRequesterId(String requesterId) {
        this.requesterId = requesterId;
    }

    public String getShelterOwnerId() {
        return shelterOwnerId;
    }

    public void setShelterOwnerId(String shelterOwnerId) {
        this.shelterOwnerId = shelterOwnerId;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Pet getPet() {
        return pet;
    }

    public void setPet(Pet pet) {
        this.pet = pet;
    }

    public User getRequester() {
        return requester;
    }

    public void setRequester(User requester) {
        this.requester = requester;
    }
}

File: Frontend/lib/models/adoption_request.dart
import 'package:pet_connect_app/models/pet.dart';
import 'package:pet_connect_app/models/user.dart';

class AdoptionRequest {
  final int id;
  final int petId;
  final String requesterId;
  final String shelterOwnerId;
  final String status;
  final Pet pet;
  final User requester;

  AdoptionRequest({
    required this.id,
    required this.petId,
    required this.requesterId,
    required this.shelterOwnerId,
    required this.status,
    required this.pet,
    required this.requester,
  });

  factory AdoptionRequest.fromJson(Map<String, dynamic> json) {
    return AdoptionRequest(
      id: json['id'],
      petId: json['petId'],
      requesterId: json['requesterId'],
      shelterOwnerId: json['shelterOwnerId'],
      status: json['status'],
      pet: Pet.fromJson(json['pet']),
      requester: User.fromJson(json['requester']),
    );
  }
}
